DietPi-Software | Minor quoting and coding enhancements

MichaIng opened the pull request:
+ DietPi-Software | p7zip-full is DietPi core dependency, thus does not need to be installed and should not be removed on desktop uninstall

Other minor changes in aim to:
- Always use double quotes when assigning value to variable, at least if it is not known to be a integer value.
- Use single quotes, if no variable or escaping used in string.
- Allow leading spaces when reading/setting dietpi.txt or config file generally.
- Prefer "`grep <pattern> <filename>`" over "`cat <filename> | grep <pattern`": https://github.com/Fourdee/DietPi/issues/1510

Commit:
v6.3

+ DietPi-Software | Minor quoting and coding rearrangements
+ DietPi-Software | p7zip-full is DietPi core dependency, thus does not need to be installed and should not be removed on desktop uninstall

Commit:
v6.3

+ DietPi-Software | Coding: Prefer "grep <pattern> <filename>" over "cat <filename> | grep <pattern"

Fourdee commented:
@MichaIng 

> Minor

That is alot of changes, looking good though, great work 👍

Commit:
v6.3

+ Whoopsie, forgotten wildcard

MichaIng commented:
@Fourdee 
Hehe jep, actually I wanted to start working on the uninstall order solution. But to refresh my overview I went through the whole script with https://github.com/Fourdee/DietPi/issues/1510 and other mentioned coding style ideas in mind.
As this sums up quite much, I separated it from the actual uninstall order method, especially since the above should be unproblematic changes, where the latter is just one of several possible solutions with pros and cons to discuss 😉.

I will merge this, tested with some dietpi-software use, example installations and uninstallations. https://www.shellcheck.net/ is still loading... I am a bid afraid of the result, would be nice to make it show only "red" errors 🤣.

Btw. would be okay for me, if you want to have full commit/merge control again. But I will anyway create PR for every non-trivial or >~3 lines change to allow review. Otherwise keep an eye and mention, if there was any too fast/large commit from my side.

Commit:
v6.3

+ Tiny: Revert to gap between install state list and choice system, it looks better :)
